File: problem_052.swift
#!/usr/bin/xcrun swift
// -*- compile-command: "swiftc problem_052.swift -o problem_052" -*-
// Copyright (c) 2016 Michael Caldwell
import Cocoa

print("Project Euler - Problem 52:")
print("Find the smallest positive integer, x, such that 2x, 3x, 4x, 5x, and 6x, contain the same digits.\n")

func numSig(n:Int) -> Int {
    var n = n
    var sig = 0
    while n > 0 {
        sig += Int(pow(10.0, Float(n) % 10))
        n /= 10
    }
    return sig
}

var solution = 0
for i in 1000...1000000 {
    var aSig = numSig(i)
    solution = i
    for m in 2...6 {
        if numSig(m*i) != aSig {
            solution = 0
            break
        }
    }

    if solution != 0 {
        break
    }
}

print(solution)

File: problem_052.py
#!/usr/bin/env python

print("Project Euler - Problem 52:")
print("Find the smallest positive integer, x, such that 2x, 3x, 4x, 5x, and 6x, contain the same digits.\n")

answer = 0

for x in range(1000000):
    sig = sorted(str(x))
    for n in range(2, 7):
        if sig == sorted(str(x * n)):
            answer = x
        else:
            answer = 0
            break

    if answer != 0:
        break

print(answer)
